### PR TITLE
Migrate `verdi calculation` to use the click infrastructure

### LINKED ISSUE #1579: verdi click: migrate verdi calculation
Migrate the verdi command `calculation` to use the new `click` framework

### PR DESCRIPTION
**Pull request by sphuber**
Fixes #1579

### CODE REVIEW

**Comment by sphuber** [comment]
This PR is ready and the tests are passing. @giovannipizzi @DropD ready for review

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -6,26 +6,33 @@
...
         aiida/scheduler/.*.py|
-        .ci/test_plugin_testcase.py|
-        aiida/backends/tests/verdi_commands.py|
+        aiida/transport/.*.py|
         aiida/orm/data/cif.py|
         aiida/restapi/resources.py|
         aiida/restapi/translator/data/cif.py|
-        aiida/utils/fixtures.py
+        aiida/utils/fixtures.py|
+        docs/update_req_for_rtd.py| # a|b -> match a OR b
```
Comment (giovannipizzi) [inline .pre-commit-config.yaml]: I'm not sure it should finish with a vertical pipe `|`

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -38,22 +45,29 @@
...
-        .ci/test_plugin_testcase.py|
-        aiida/backends/tests/verdi_commands.py|
+        aiida/control/.*.py|
         aiida/orm/data/cif.py|
-        aiida/restapi/translator/data/cif.py|
         aiida/restapi/resources.py|
-        aiida/utils/fixtures.py
+        aiida/restapi/translator/data/cif.py|
+        aiida/utils/fixtures.py|
+        docs/update_req_for_rtd.py|
```
Comment (giovannipizzi) [inline .pre-commit-config.yaml]: Same here

File: .pre-commit-config.yaml (hunk 1 above)
Comment (sphuber) [reply .pre-commit-config.yaml]: Only for that line, or for all of them?

File: aiida/orm/implementation/general/calculation/job/__init__.py [hunk 3]
```diff
@@ -50,7 +50,6 @@ class JobCalculationFinishStatus(enum.Enum):
     RETRIEVALFAILED = 200
     PARSINGFAILED = 300
     FAILED = 400
-    I_AM_A_TEAPOT = 418
```
Comment (giovannipizzi) [inline aiida/orm/implementation/general/calculation/job/__init__.py]: Interesting old state ;-)

File: aiida/transport/plugins/test_local.py [hunk 4]
```diff
@@ -48,6 +48,17 @@ def test_basic(self):
         with LocalTransport():
             pass
 
+    def test_is_open(self):
```
Comment (giovannipizzi) [inline aiida/transport/plugins/test_local.py]: This test should probably be defined in `test_all_plugins.py` with the `@run_for_all_plugins` decorator?

File: aiida/transport/plugins/test_ssh.py [hunk 5]
```diff
@@ -57,6 +57,17 @@ def test_no_host_key(self):
         # Reset logging level
         logging.disable(logging.NOTSET)
 
+    def test_is_open(self):
```
Comment (giovannipizzi) [inline aiida/transport/plugins/test_ssh.py]: Same comment as the analogous test in `test_local.py`

File: .pre-commit-config.yaml (hunk 1 above)
Comment (giovannipizzi) [reply .pre-commit-config.yaml]: If I understand the syntax, the | is an "OR" so it should be everywhere except the last line (btw I'm still not convinced that the syntax is perfectly ok - e.g. the dot probably matches any character?)

Comment (sphuber) [reply .pre-commit-config.yaml]: Ah, yes I did not recognize that you meant the last line. I think you are right and I removed it. The dot should indeed match any character. I tried removing it before the wildcard and then it does not match anything. It really seems to work like a regex pattern then

File: aiida/orm/implementation/general/calculation/job/__init__.py (hunk 3 above)
Comment (sphuber) [reply aiida/orm/implementation/general/calculation/job/__init__.py]: Yeah, I put that in when it was still the `workflows` branch. Thought it best to just remove it now before someone gets hurt :)

**Pull request merged**